js/components/closeButton.js

```diff
@@ -1,14 +1,23 @@
 var React = require('react'),
-    Icon = require('../components/icon');
+    Icon = require('../components/icon'),
+    Utils = require('./utils'),
+    CONSTANTS = require('../constants/constants');
 
 var CloseButton = React.createClass({
   render: function() {
     return (
-        <button className={this.props.cssClass}
-          onClick={this.props.closeAction}>
-          <Icon {...this.props} icon="dismiss"
-            className={this.props.className}/>
-        </button>
+      <button
+        className={this.props.cssClass}
+        onClick={this.props.closeAction}
+        onMouseUp={Utils.blurOnMouseUp}
+        data-focus-id={'close-' + Date.now()}
```

Comment:
Why do we need to add Date.now() to the data-focus-id?

Reply:
This is in case we ever display more than one close button at a time, the id needs to be unique.

Reply:
Just out of curiosity, what happens when the id's for the close button are not unique? Let's say the user changed their system clock between viewing two different close buttons, and now two close buttons have the same data-focus-id (however harder-than-winning-every-lottery-in-the-world this may be) ?

Reply:
Nothing in this case since this is mostly used for control bar elements, but the idea is that you use it to store and find the element that needs to be refocused after the control bar is re-rendered, but if there are many elements with the same id you will likely get the wrong one when querying the DOM.